### DISCUSSION TITLE
kubernetes-handbook/objects/pod/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/cloud-native-library#77] [Announcements]
# kubernetes-handbook/objects/pod/

Pod 是 Kubernetes 中可以创建的最小部署单元，也是 Kubernetes REST API 中的顶级资源类型。 在 Kuberentes V1 core API 版本中的 Pod 的数据结构如下图所示： Pod Cheatsheet 什么是 Pod？ Pod 就像是豌豆荚一样，它由一个或者多个容器组成（例如 Docker 容器），它们共享容器存储、网

https://lib.jimmysong.io/kubernetes-handbook/objects/pod/

**Comment by GingJan** [comment]
`当 Pod 因为某种原因被删除或者被新创建的相同的 Pod 取代，它相关的东西（例如 volume）也会被销毁和再创建一个新的 volume `
和
`资源共享和通信`的`Volume 能够保证 pod 重启时使用的数据不丢失`
这两句话是否相互矛盾，前者说一同销毁，后者却说volume的数据不丢失？

**Reply by rootsongjc** [reply]
文中确实前后矛盾，没有说清楚 Volume 的类型，当 Pod 删除或重启后 Volume 中的数据是否会丢失，这取决于 Volume 的类型。临时卷类型的生命周期与 Pod 相同，但持久卷可以比 Pod 的存活期长。详见 [Kubernetes 文档](https://kubernetes.io/zh-cn/docs/concepts/storage/volumes/)。
